src/Checker/AssignByReferenceVisitor.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+
+namespace umulmrum\PhpReferenceChecker\Checker;
+
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\AssignRef;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\NodeVisitorAbstract;
+use umulmrum\PhpReferenceChecker\DataModel\MethodRepository;
+use umulmrum\PhpReferenceChecker\DataModel\NonReferenceAssignmentWarning;
+
+class AssignByReferenceVisitor extends NodeVisitorAbstract
+{
+
+    /**
+     * @var array
+     */
+    private $warnings;
+    /**
+     * @var MethodRepository
+     */
+    private $repository;
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @param array            $warnings
+     * @param MethodRepository $repository
+     * @param string           $path
+     */
+    public function __construct(array &$warnings, MethodRepository $repository, $path)
+    {
+        $this->warnings =& $warnings;
+        $this->repository = $repository;
+        $this->path = $path;
+    }
+
+    public function enterNode(Node $node)
+    {
+        parent::enterNode($node);
+
+        if (false === $node instanceof AssignRef) {
+            return null;
```

Comment:
The method doesn't have a return value - please "return" only (and add a docblock for the method as the parent fails to do so).